src/builder.test.ts

```diff
@@ -117,11 +117,14 @@ describe("ArrayResult.grab/UnknownResult.grab/EntityResult.grab", () => {
     expect(data[1]).toEqual({ _id: "pokemon.2" });
     expect(data[3]).toEqual({ _id: "pokemon.4", name: "Charmander", hp: 39 });
 
+    type pokemonName = InferType<typeof pokemonQuery>[number]["name"];
+
     for (const dat of data) {
       if ("name" in dat && dat.name === "Charmander") {
-        expect(dat.name === "Charmander").toBeTruthy();
+        const name: pokemonName = dat.name;
```

Comment:
Would this change have caused it to fail with the previously implemented type?

Reply:
yeaup. previously `"name"` wouldn't be indexible on the type `Pokemon` because the base pokemon shape (`{ _id: string }`) would be part of the union.

At the same time tho, @gksander brought up an interesting point in slack.. if it's conditional selection, there's a chance no conditions are met and you're left with and entity that's the shape of the base properties